packages/collaboration-manager/src/CollaborationManager.ts

```diff
@@ -11,16 +13,68 @@ export class CollaborationManager {
    */
   #model: EditorJSModel;
 
+  /**
+   * UndoRedoManager instance to manage undo/redo operations
+   */
+  #undoRedoManager: UndoRedoManager;
+
+  /**
+   * Flag to control whether events should be handled to avoid putting operations to the stack on undo/redo
+   */
+  #shouldHandleEvents = true;
+
+
   /**
    * Creates an instance of CollaborationManager
    *
    * @param model - EditorJSModel instance to listen to and apply operations
    */
   constructor(model: EditorJSModel) {
     this.#model = model;
+    this.#undoRedoManager = new UndoRedoManager();
     model.addEventListener(EventType.Changed, this.#handleEvent.bind(this));
   }
 
+  /**
+   * Undo last operation in the local stack
+   */
+  public undo(): void {
+    const operation = this.#undoRedoManager.undo();
+
+    if (operation === undefined) {
+      return;
+    }
+
+    // Disable event handling
+    this.#shouldHandleEvents = false;
+
+    this.applyOperation(operation);
+    this.#undoRedoManager.putToRedoStack(Transformer.inverse(operation));
```

Comment:
This should be inside of the undoRedoManager

So undo method would
1. Pop operation from undo stack
2. Inverse operation
3. Put into redo stack
4. Return it back to CollaborationManager